lib/ts/recipe/totp/api/implementation.ts

```diff
@@ -0,0 +1,99 @@
+/* Copyright (c) 2023, VRAI Labs and/or its affiliates. All rights reserved.
+ *
+ * This software is licensed under the Apache License, Version 2.0 (the
+ * "License") as published by the Apache Software Foundation.
+ *
+ * You may not use this file except in compliance with the License. You may
+ * obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+import { APIInterface } from "../types";
+import TotpRecipe from "../recipe";
+import SessionError from "../../session/error";
+// import { MfaClaim, completeFactorInSession } from '../../mfa';
+
+export default function getAPIImplementation(): APIInterface {
+    return {
+        createDevicePOST: async function ({ session, options, deviceName, userContext }) {
+            let userIdentifierInfo: string | undefined = undefined;
+            const emailOrPhoneInfo = await TotpRecipe.getInstanceOrThrowError().getUserIdentifierInfoForUserId(
+                session.getUserId(),
+                userContext
+            );
+            if (emailOrPhoneInfo.status === "OK") {
+                userIdentifierInfo = emailOrPhoneInfo.info;
+            } else if (emailOrPhoneInfo.status === "UNKNOWN_USER_ID_ERROR") {
+                throw new SessionError({
+                    type: SessionError.UNAUTHORISED,
+                    message: "Unknown User ID provided",
+                });
+            }
+
+            let existingDeviceCount = 0;
+            let verifiedDeviceCount = 0;
+            if (deviceName === undefined) {
```

Comment:
- The device name generation should happen in the core, so remove the logic here. I see no benefit of doing it in the backend SDK level.
- See last comment in https://github.com/supertokens/supertokens-node/pull/576#discussion_r1212621133